src/main/kotlin/screeps/api/PowerCreep.kt

```diff
@@ -0,0 +1,59 @@
+package screeps.api
+
+import screeps.api.structures.Structure
+import screeps.api.structures.StructureController
+import screeps.api.structures.StructurePowerBank
+import screeps.api.structures.StructurePowerSpawn
+
+abstract external class PowerCreep: RoomObject, Owned, Attackable, Identifiable {
+    companion object {
+        fun create(name: String, className: PowerClassConstant): ScreepsReturnCode
+    }
+
+    val carry: StoreDefinition
+    val carryCapacity: Int
+    val className: PowerClassConstant
+    val deleteTime: Int?
+    val level: Int
+    val memory: PowerCreepMemory
+    val name: String
+    //val powers // Not sure how to do this
```

Comment:
This could be modeled as 
```suggestion

    external interface Powers {
         level: Int
         cooldown: Int?
    }
    val powers : Powers // Not sure how to do this
```